### PR TITLE
lib/promscrape: fixed reload on max_scrape_size change

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
### Describe Your Changes

fixed reload on max_scrape_size change
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7260

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Review by f41gh7 — APPROVED** [review]
LGTM

File: docs/changelog/CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ See also [LTS releases](https://docs.victoriametrics.com/lts-releases/).
 * BUGFIX: [vmgateway](https://docs.victoriametrics.com/vmgateway/): fix possible panic during parsing of a token without `vm_access` claim. This issue was introduced in v1.104.0.
 * BUGFIX: [vmui](https://docs.victoriametrics.com/#vmui): fix error messages rendering from overflowing the screen with long messages. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7207).
 * BUGFIX: [vmctl](https://docs.victoriametrics.com/vmctl/): properly add metrics tags for `opentsdb` migration source. Previously it could have empty values. See [this PR](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7161).
+* BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent/): fixed hot-reload on `max_scrape_size` change. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7260).
```
Comment (hagen1778) [inline docs/changelog/CHANGELOG.md:35]:
```suggestion
* BUGFIX: [Single-node VictoriaMetrics](https://docs.victoriametrics.com/)  and [vmagent](https://docs.victoriametrics.com/vmagent/): properly update `max_scrape_size` param change during [hot-reload](https://docs.victoriametrics.com/vmagent/#configuration-update). See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/7260).
```

Comment (hagen1778) [reply docs/changelog/CHANGELOG.md:35]: @AndrewChubatiuk @f41gh7 please pay more attention to the changelog lines. Their main purpose is to be clear for readers. I've updated https://docs.victoriametrics.com/contributing/#pull-request-checklist to highlight this requirement:
> A line in the [changelog](https://docs.victoriametrics.com/changelog/#tip) mentioning the change and related issue in a way that would be clear to other readers even if they don't have the full context.

**Comment by hagen1778** [comment]
It would be great to add tests that could verify the change is effective.

**Comment by jiekun** [comment]
> It would be great to add tests that could verify the change is effective.

how about writing a test that **fail every time** when new param/arg/... are introduced to `ScrapeWork`. (might be complex)

Alternatively, simply add an `TODO` or `notice` to `type ScrapeWork struct` may help. 
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/eefae854500e742b4788c16cedd883e20f335a3d/lib/promscrape/scrapework.go#L46

**Review by hagen1778 — APPROVED** [review]
Thanks!

**Pull request merged**